src/transformers/generation/utils.py

```diff
@@ -1785,6 +1790,16 @@ def typeerror():
                 **model_kwargs,
             )
 
+        if generation_config.cache_implementation in NEED_SETUP_CACHE_CLASSES_MAPPING:
+            if not callable(getattr(self, "_reset_cache", None)):
+                raise ValueError(
+                    "The `generation_config` defines a `cache_implementation` that is not compatible with this model."
+                    " Make sure it has a `_reset_cache` function."
+                )
+            self._reset_cache()
```

Comment:
Feel likes a memory leak if we don't have that no?

Reply:
Mmm, yes if we don't call the next `generate` with a static cache. 
Let's give a better warning: 
```suggestion
        if generation_config.cache_implementation in NEED_SETUP_CACHE_CLASSES_MAPPING:
            if not callable(getattr(self, "_reset_cache", None)):
                raise ValueError(
                    "A `static_cache` was used to generate but there was a failure when trying to  release the cache. "
                    " Make sure this model implements a `_reset_cache` function."
                )
            self._reset_cache()
```